services/src/main/java/org/fao/geonet/api/records/MetadataWorkflowApi.java

```diff
@@ -692,8 +697,12 @@ public List<MetadataStatusResponse> getWorkflowStatusByType(
         Integer size,
         HttpServletRequest request) throws Exception {
         ServiceContext context = ApiUtils.createServiceContext(request);
+        checkUserProfileToViewMetadataHistory(context.getUserSession());
 
         Profile profile = context.getUserSession().getProfile();
+        String allowedProfileLevel = org.apache.commons.lang.StringUtils.defaultIfBlank(settingManager.getValue(Settings.METADATA_HISTORY_ACCESS_LEVEL), Profile.Editor.toString());
```

Comment:
You have this same line in checkUserProfileToViewMetadataHistory()

I wonder if you should really call checkUserProfileToViewMetadataHistory()

Maybe simply to just add the check below

```
 // Is the user profile higher than the profile allowed?
            if (!UserUtil.hasHierarchyRole(allowedUserProfileToImportMetadata, this.roleHierarchy)) {
                throw new NotAllowedException("The user has no permissions to view metadata history.");
            }
```

Reply:
I think we can get rid of it. The canView and canEdit for the configured profile level is good enough. The code was copy pasted. Let me make some change for it